### PR TITLE
fix(types): fix and clean up types for request.ts

### PR DESCRIPTION
**Pull request by AVaksman**
Towards #460

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)

### CODE REVIEW

File: src/request.ts [hunk 1]
```diff
@@ -516,7 +523,7 @@ class DatastoreRequest {
   }
 
   insert(entities: Entities): Promise<CommitResponse>;
-  insert(entities: Entities, callback: CallOptions): void;
+  insert(entities: Entities, callback: CommitCallback): void;
```
Comment (callmehiphop) [inline src/request.ts]: Should we follow suit with the other commit convenience methods and make a custom type for the callback/response here that extends the commit version?

File: src/request.ts [hunk 2]
```diff
@@ -1330,45 +1325,53 @@ export type ProjectId = string | null | undefined;
...
-}
-export interface RunQueryStreamOptions {
-  gaxOptions?: CallOptions;
-  consistency?: 'strong' | 'eventual';
+  projectId?: ProjectId;
+  query?: QueryProto;
 }
-export interface SaveCallback {
-  (a?: Error | null, b?: Entity): void;
+export interface RunQueryStreamOptions extends RunQueryOptions {}
```
Comment (callmehiphop) [inline src/request.ts]: Any reason why we wouldn't just re-use `RunQueryOptions`?

File: src/request.ts [hunk 3]
```diff
@@ -1330,45 +1325,53 @@ export type ProjectId = string | null | undefined;
...
+export interface UpsertCallback extends CommitCallback {}
+export type UpsertResponse = CommitResponse;
 
 /*! Developer Documentation
  *
  * All async methods (except for streams) will return a Promise in the event
  * that a callback is omitted.
  */
-promisifyAll(DatastoreRequest);
+promisifyAll(DatastoreRequest, {exclude: ['request_']});
```
Comment (callmehiphop) [inline src/request.ts]: Do we need this change? I thought that it excluded methods with trailing underscores by default?

Comment (AVaksman) [reply src/request.ts]: You are right, forgot about it. 
Not needed!

File: src/request.ts (hunk 2 above)
Comment (AVaksman) [reply src/request.ts]: Just to keep the same "aesthetic" look as with `Save`/`Update`/`UpsertCallback` interfaces extend `CommitCallback` interface.
Not particular about it. Can change ether way?

File: src/request.ts (hunk 1 above)
Comment (AVaksman) [reply src/request.ts]: I think yes, nice catch!
Added `InsertCallback`/`Response` and `DeleteCallback`/`Response`

**Comment by AVaksman** [comment]
Thank you for reviewing. Took care of the questions

File: src/index.ts [hunk 4]
```diff
@@ -374,7 +374,7 @@ const urlSafeKey = new entity.URLSafeKey();
  * });
  */
 class Datastore extends DatastoreRequest {
-  clients_: Map<string, {}>;
+  clients_: Map<string, {[k: string]: Function}>;
```
Comment (callmehiphop) [inline src/index.ts]: You could use `import {ClientStub} from 'google-gax';` here

File: src/request.ts (hunk 2 above)
Comment (callmehiphop) [reply src/request.ts]: No need, I was mostly curious!

File: src/request.ts [hunk 5]
```diff
@@ -1330,38 +1323,50 @@ export type ProjectId = string | null | undefined;
 export interface RequestCallback {
   (
     a?: Error | null,
-    b?: AllocateIdsRequestResponse &
+    b?: google.datastore.v1.IAllocateIdsResponse &
```
Comment (callmehiphop) [inline src/request.ts]: Hopefully I'm not opening a big can of worms, but should these response types be listed in this callback? Seems like maybe this should be a generic type? WDYT?

File: src/index.ts (hunk 4 above)
Comment (AVaksman) [reply src/index.ts]: Nice! Done
Thanks

File: src/request.ts (hunk 5 above)
Comment (AVaksman) [reply src/request.ts]: It should, it is internal anyways.

Comment (AVaksman) [reply src/request.ts]: I did use `any` vs `Entity` which is masked `any` or vs extending from `GetCallback`.
I think it gives better code readability as "what you see is what you get" (personal preference).

Comment (callmehiphop) [reply src/request.ts]: Sounds good to me!

**Pull request merged**